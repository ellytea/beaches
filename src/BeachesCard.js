import React, { Component } from 'react';
import './App.css';


class BeachesCard extends Component {
  constructor(props) {
    super();
  }

  render() {
    const beachProps = this.props.selectedBeach ? Object.keys(this.props.selectedBeach) : []
    if (!this.props.selectedBeach) {
      return ( <div className='none'></div> );
    } else {
    return (
      <div className='popup'>
        <img className='popup-img' src={this.props.selectedBeach.image}/>
        <h1>{this.props.selectedBeach.name}</h1>
        <p>{this.props.selectedBeach.county}</p>
        <ul>Recommended Activities:
           {
            this.props.selectedBeach.activities.map((activity) => {
              return <li>{activity}</li>
            })
           }
         </ul>
         <ul>Amenities:
           {
            beachProps.map((prop) => {
               if (this.props.selectedBeach[prop] === true) {
                 return <li>{prop}</li>
               }
            })
           }
        </ul>
        <button className='pop-btn' onClick={this.props.closeBeach}>X close</button>
      </div>
    )
    }
  }
  
    

}


export default BeachesCard;
