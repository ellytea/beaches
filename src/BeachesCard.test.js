import React from 'react';
import BeachesCard from './BeachesCard.js';
import { shallow } from 'enzyme';

const closeBeachMock = jest.fn();

describe('BeachesCard', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <BeachesCard closeBeach={closeBeachMock} />
      )
  })

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })


})