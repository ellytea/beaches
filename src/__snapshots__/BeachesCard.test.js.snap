// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeachesCard should match the snapshot with all data passed in 1`] = `
<div
  className="none"
/>
`;